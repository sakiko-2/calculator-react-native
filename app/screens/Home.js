import React, { Component } from 'react';
import { SafeAreaView, StyleSheet, View } from 'react-native';

import Display from '../components/Display';
import Button from '../components/Button';
import ButtonRow from '../components/ButtonRow';

class Home extends Component {
  state = {
    current: '0',
    previous: null,
    operator: null,
    showDisplay: true
  };

  handlePress(value) {
    if (typeof value === 'number') {
      this.handleNumber(value);
    }
    else if (typeof value === 'string') {
      this.handleOperator(value);
    }
  };

  handleNumber(num) {
    this.setState(state => {
      const { current } = state;
      if (current === '0') {
        return {
          current: `${num}`,
          showDisplay: true
        }
      } 
      else if (current.replace('.', '').length > 8) {
        return {
          current: `${current}`,
          showDisplay: true
        }
      }
      else {
        return {
          current: `${current}${num}`,
          showDisplay: true
        }
      }
    });
  };

  handleOperator(str) {
    const { current, previous, operator } = this.state;
    const curr = parseFloat(current);
    const prev = parseFloat(previous);

    switch (str) {
      case 'c':
        this.setState(
          defaultState
        );
        break;
      case '+/-': 
        this.setState({
          current: `${curr * -1}`,
          showDisplay: true
        });
        break;
      case '%':
        this.setState({
          current: `${curr / 100}`,
          showDisplay: true
        });
        break;
      case '/':
      case '*':
      case '-':
      case '+':
        this.setState({
          showDisplay: false,
          operator: str,
          previous: current,
          current: '0'
        });
        break;
      case '.':
        this.setState({
          current: current.includes('.') ? `${current}` : `${current}.`,
          showDisplay: true
        });
        break;
      case '=':
        this.setState({
          operator: null,
          previous: 0,
          current: current && operator ? eval(prev + operator + curr).toString() : `${current}`,
          showDisplay: true
        });
        break;
    }
  };

  formatNumber(num) {
    return parseFloat(num).toLocaleString('en-US', { maximumFractionDigits: 8 });
  };

  render() {
    const darkColor = '#444';
    const lightColor = '#9e9e9e';
    const accentColor = '#ffa500';
    const lightText = '#fff';
    const darkText = '#000';
    const { current, previous, showDisplay } = this.state;
    const display = showDisplay ? current : previous;

    return (
      <SafeAreaView>
        <View style={styles.container}>
          <Display text={this.formatNumber(display)}/>
          <ButtonRow>
            <Button text={'C'} color={darkText} backgroundColor={lightColor} onPress={() => this.handlePress('c')} />
            <Button text={'+/-'} color={darkText} backgroundColor={lightColor} onPress={() => this.handlePress('+/-')} />
            <Button text={'%'} color={darkText} backgroundColor={lightColor} onPress={() => this.handlePress('%')} />
            <Button text={'รท'} color={lightText} backgroundColor={accentColor} onPress={() => this.handlePress('/')} />
          </ButtonRow>
          <ButtonRow>
            <Button text={'7'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(7)} />
            <Button text={'8'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(8)} />
            <Button text={'9'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(9)} />
            <Button text={'x'} color={lightText} backgroundColor={accentColor} onPress={() => this.handlePress('*')} />
          </ButtonRow>
          <ButtonRow>
            <Button text={'4'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(4)} />
            <Button text={'5'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(5)} />
            <Button text={'6'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(6)} />
            <Button text={'-'} color={lightText} backgroundColor={accentColor} onPress={() => this.handlePress('-')} />
          </ButtonRow>
          <ButtonRow>
            <Button text={'1'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(1)} />
            <Button text={'2'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(2)} />
            <Button text={'3'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress(3)} />
            <Button text={'+'} color={lightText} backgroundColor={accentColor} onPress={() => this.handlePress('+')} />
          </ButtonRow>
          <ButtonRow>
            <Button text={'0'} color={lightText} backgroundColor={darkColor} wide='true' onPress={() => this.handlePress(0)} />
            <Button text={'.'} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress('.')} />
            <Button text={'='} color={lightText} backgroundColor={darkColor} onPress={() => this.handlePress('=')} />
          </ButtonRow>
        </View>
      </SafeAreaView>
    );
  };
};

const defaultState = {
  current: '0',
  previous: null,
  operator: null,
  showDisplay: true
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#000',
    padding: 10
  }
});

export default Home;
