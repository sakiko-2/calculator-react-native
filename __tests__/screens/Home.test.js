import 'react-native';
import React from 'react';
import Home from '../../app/screens/Home';

import renderer from 'react-test-renderer';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
