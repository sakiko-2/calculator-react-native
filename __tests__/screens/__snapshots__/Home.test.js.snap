// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#000",
          "flex": 3,
          "justifyContent": "flex-end",
          "padding": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 80,
            "textAlign": "right",
          }
        }
      >
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#9e9e9e",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            C
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#9e9e9e",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            +/-
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#9e9e9e",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            %
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#ffa500",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            รท
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            7
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            8
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#ffa500",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            x
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            4
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            5
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#ffa500",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            2
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#ffa500",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
              Object {
                "alignItems": "flex-start",
                "flex": 2,
                "paddingLeft": 79.75,
                "width": 365,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            .
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222",
                "borderRadius": 187.5,
                "flex": 1,
                "height": 177,
                "justifyContent": "center",
                "margin": 5,
                "width": 177,
              },
              Object {
                "backgroundColor": "#444",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 32,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
